src/register/registrationFields/CountryField.jsx

```diff
@@ -1,154 +1,62 @@
+import React from 'react';
 
 import { useIntl } from '@edx/frontend-platform/i18n';
-import { Icon, IconButton } from '@edx/paragon';
-import { ExpandLess, ExpandMore } from '@edx/paragon/icons';
+import { FormAutosuggest, FormAutosuggestOption, FormControlFeedback } from '@edx/paragon';
 import PropTypes from 'prop-types';
 
-import { FormGroup } from '../../common-components';
-import {
-  COUNTRY_CODE_KEY, COUNTRY_DISPLAY_KEY, EXPAND_LESS_ICON, EXPAND_MORE_ICON,
-} from '../data/constants';
+import { COUNTRY_CODE_KEY, COUNTRY_DISPLAY_KEY } from '../data/constants';
 import messages from '../messages';
 
 const CountryField = (props) => {
   const { countryList, selectedCountry } = props;
-
-  const dropdownRef = useRef(null);
   const { formatMessage } = useIntl();
-  const [errorMessage, setErrorMessage] = useState(props.errorMessage);
-  const [dropDownItems, setDropDownItems] = useState([]);
-  const [displayValue, setDisplayValue] = useState('');
-  const [trailingIcon, setTrailingIcon] = useState(EXPAND_MORE_ICON);
 
-  const onBlurHandler = (event, itemClicked = false, countryName = '') => {
-    const { name } = event.target;
-    const relatedName = event.relatedTarget ? event.relatedTarget.name : '';
-    // For a better user experience, do not validate when focus out from 'country' field
-    // and focus on 'countryItem' or 'countryExpand' button.
-    if ((relatedName === 'countryItem' || relatedName === 'countryExpand') && name === 'country') {
-      return;
-    }
-    const countryValue = itemClicked ? countryName : displayValue;
-    if (props.onBlurHandler) {
-      props.onBlurHandler({ target: { name: 'country', value: countryValue } });
-    }
-    setTrailingIcon(EXPAND_MORE_ICON);
-    setDropDownItems([]);
-  };
-
-  const getDropdownItems = (countryToFind = null) => {
-    let updatedCountryList = countryList;
-    if (countryToFind) {
-      updatedCountryList = countryList.filter(
-        (option) => (option.name.toLowerCase().includes(countryToFind.toLowerCase())),
-      );
-    }
-
-    return updatedCountryList.map((country) => {
-      const countryName = country[COUNTRY_DISPLAY_KEY];
-      return (
-        <button
-          type="button"
-          name="countryItem"
-          className="dropdown-item data-hj-suppress"
-          value={countryName}
-          key={country[COUNTRY_CODE_KEY]}
-          onClick={(event) => onBlurHandler(event, true, countryName)}
-          /* This event will prevent the blur event to be fired,
-           as blur event is having higher priority than click event and restricts the click event.
-          */
-          onMouseDown={(event) => event.preventDefault()}
-        >
-          {countryName.length > 30 ? countryName.substring(0, 30).concat('...') : countryName}
-        </button>
-      );
-    });
+  const handleSelected = (value) => {
+    if (props.onBlurHandler) { props.onBlurHandler({ target: { name: 'country', value } }); }
```

Comment:
please destructure the onBlurHandler from props here too

Reply:
The reason I did not destructure the handlers here was the naming of handlers and functions of components. the `onFocusHandler` and `onChangeHandler` are of the same names. So, I just kept all handlers inside the `props`. 
If we are to destructure them from props, we need to change the names of component functions.

But if you still feel like its worth destructing and we should change the component's function names. I will be happy to do that.

Reply:
its fine we can keep it